### ISSUE TITLE
Instance overloaded: failed to create OS thread: Resource temporarily unavailable

### ISSUE THREAD

**Issue by UPetit** [PrefectHQ/prefect#4148]
## Description
I've ran the prefect server using `prefect server start` yesterday night on an Ubuntu 18.04 instance (3x x86 64bit, 4GB Mem).
All the containers started one after the other correctly.
3 hours later too many processes were running in parallel on my instance (please note that nothing else is running in parallel except from these containers) and the **hasura container exited**.
- On the **hasura** container:
```
graphql-engine: failed to create OS thread: Resource temporarily unavailable
```
Before crashing, graphql container was pinging an IP all the time and it sounds quite related to the increase of the pids it took over time before crashing.
- On the **postgres** container:
```
2021-02-22 00:43:39.683 UTC [1] LOG:  could not fork autovacuum worker process: Resource temporarily unavailable
```
- On the **towel** container (server core):
```
{"severity": "INFO", "name": "prefect-server.Scheduler", "message": "Scheduled 0 flow runs."}
{"severity": "DEBUG", "name": "prefect-server.Scheduler", "message": "Heartbeat from Scheduler: next run at 2021-02-22T08:24:17+00:00"}
{"severity": "DEBUG", "name": "prefect-server.Lazarus", "message": "Heartbeat from Lazarus: next run at 2021-02-22T08:29:17+00:00"}
{"severity": "ERROR", "name": "prefect-server.ZombieKiller", "message": "Unexpected error: ConnectError(gaierror(-2, 'Name or service not known'))"}
{"severity": "DEBUG", "name": "prefect-server.ZombieKiller", "message": "Heartbeat from ZombieKiller: next run at 2021-02-22T08:23:25+00:00"}
{"severity": "ERROR", "name": "prefect-server.ZombieKiller", "message": "Unexpected error: ConnectError(gaierror(-2, 'Name or service not known'))"}
```
On the **apollo** container (nothing special):
```
Server ready at http://0.0.0.0:4200 🚀 (version: 2021.02.12)
Sending telemetry to Prefect Technologies, Inc.: {"source":"prefect_server","type":"heartbeat","payload":{"id":"9ae554bd-8f86-4785-ab82-0ab41eb64130","prefect_server_version":"2021.02.12","api_version":"0.2.0"}}
```

All instances except from the hasura on are "up" (if we can say so...) but of course the UI has blank body.
![Screenshot from 2021-02-22 12-33-18](https://user-images.githubusercontent.com/22870826/108702898-2a049c00-750a-11eb-98c4-8bab4f630e5c.png)

It happened to me yesterday morning first but I believed it was due to other programs running on the instance so I rebooted the instance and just started the prefect server.
It sounds like too many processes are running in parallel, it's overloaded, any command in my bash terminal is randomly executed because there is no available process:
`bash: fork: retry: Resource temporarily unavailable`

## Reproduction
I just started the server without even using it after. It just ended up with a lack of available processes it seems.

## Environment
Ubuntu 18.04 (3x x86 64bit, 4GB Mem)

I can't run `prefect diagnostics` because of the overloading of the instance I'm sorry.

Hope you can help me.
Thank you in advance

EDIT: I've restarted my instance, even if it's temporarily working (until the expected crash), here is the diagnostics:
```
{
  "config_overrides": {},
  "env_vars": [
    "PREFECT__CONTEXT__SECRETS__GITHUB_ACCESS_TOKEN"
  ],
  "system_information": {
    "platform": "Linux-4.15.0-118-generic-x86_64-with-glibc2.27",
    "prefect_backend": "server",
    "prefect_version": "0.14.9",
    "python_version": "3.8.7"
  }
}
```
EDIT2: I've looked at the pids of each container and apollo and graphql ones keep increasing like crazy (one by one but it doubled in one minute from 150 to 300...). Here is what it looks like in DockStation:
![Screenshot from 2021-02-22 12-57-05](https://user-images.githubusercontent.com/22870826/108705379-84ebc280-750d-11eb-84ab-402372d47881.png)

**Comment by madkinsz** [comment]
Thanks for the details @UPetit -- this is a duplicate of #4114 so I'm going to close it in favor of that. This should be resolved soon.
